### PR TITLE
fix: Return nil if negative time elapsed from last block happens

### PR DESCRIPTION
**Pull request by RafilxTenfen**
## Description

Implemented Option 3: Modify AccrueAllInterest to treat "negative time elapsed" as "zero time elapsed", aborting the function without modifying the state until an EndBlock where BlockTime > LastInterestTime.

closes: #543 

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/leverage/keeper/interest.go [hunk 1]
```diff
@@ -73,7 +73,13 @@ func (k Keeper) AccrueAllInterest(ctx sdk.Context) error {
...
 	// calculate time elapsed since last interest accrual (measured in years for APR math)
 	if currentTime < prevInterestTime {
-		return types.ErrNegativeTimeElapsed.Wrap(fmt.Sprintf("current: %d, prev: %d", currentTime, prevInterestTime))
+		k.Logger(ctx).With("AccrueAllInterest will wait for block time > prevInterestTime").Error(
+			types.ErrNegativeTimeElapsed.Error(),
+			"current", currentTime,
+			"prev", prevInterestTime,
+		)
+
+		return nil
```
Comment (robert-zaremba) [inline x/leverage/keeper/interest.go:82]: Log is not enough. I think we should either return some message or keep the error.

Comment (RafilxTenfen) [reply x/leverage/keeper/interest.go:82]: What do you think about [these options](https://github.com/umee-network/umee/issues/543#issuecomment-1152706616)?

**Review by toteki — APPROVED** [review]

**Comment by RafilxTenfen** [comment]
@robert-zaremba So, do you think we should change from `Error` to `Warn`?

**Comment by toteki** [comment]
Message can keep `Error` log level for now. Once tendermint fixes, we can return error instead of `nil`, which would bring it back to causing chain halt (which we want in this case).

**Comment by robert-zaremba** [comment]
OK

File: x/leverage/keeper/interest.go (hunk 1 above)
Comment (robert-zaremba) [reply x/leverage/keeper/interest.go:84]: OK

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**